### PR TITLE
[Issue #732] support one table scan

### PR DESCRIPTION
**Pull request by huasiy**

### CODE REVIEW

**Comment by huasiy** [comment]
Resolves #732

**Comment by huasiy** [comment]
image

**Comment by huasiy** [comment]
image

File: pixels-turbo/pixels-worker-common/src/main/java/io/pixelsdb/pixels/worker/common/BaseScanStreamWorker.java [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            WorkerThreadExceptionHandler exceptionHandler = new WorkerThreadExceptionHandler(logger);
+            ExecutorService threadPool = Executors.newFixedThreadPool(cores * 2,
+                    new WorkerThreadFactory(exceptionHandler));
+
+            String outputFolder = event.getOutput().getPath();
+            if (!outputFolder.endsWith("/")) {
+                outputFolder += "/";
+            }
+            Queue<String> outputPaths = new ConcurrentLinkedQueue<>(
+                    ScanInput.generateOutputPaths(outputFolder, 100));
```
Comment (bianhq) [inline pixels-turbo/pixels-worker-common/src/main/java/io/pixelsdb/pixels/worker/common/BaseScanStreamWorker.java:105]: Do not hardcode the numSplits.

**Review by bianhq — CHANGES_REQUESTED** [review]

File: pixels-turbo/pixels-worker-vhive/pom.xml [hunk 2]
```diff
@@ -170,29 +170,27 @@
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-assembly-plugin</artifactId>
-                <version>3.5.0</version>
+                <artifactId>maven-shade-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (huasiy) [reply pixels-turbo/pixels-worker-vhive/pom.xml:173]: Yes. Refer to issue #703.

[the remark this answers is not in the document]
Comment (bianhq) [reply pixels-turbo/pixels-worker-vhive/pom.xml:173]: OK

File: pixels-common/src/main/resources/pixels.properties [hunk 3]
```diff
@@ -188,6 +188,11 @@ executor.intra.worker.parallelism=8
 executor.function.service=lambda
 # which method will be used for data exchanging, batch or stream
 executor.exchange.method=batch
+# the first port used by pipeline worker
+executor.worker.exchange.port=50010
+# the ip and port of worker coordinator
+executor.worker.coordinator.ip=10.168.4.4
```
Comment (bianhq) [inline pixels-common/src/main/resources/pixels.properties:194]: We already have `worker.coordinate.server.host` and `worker.coordinate.server.port` in pixels.properties.

File: pixels-turbo/pixels-invoker-vhive/src/main/java/io/pixelsdb/pixels/invoker/vhive/ScanStreamInvokerProvider.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package io.pixelsdb.pixels.invoker.vhive;
+
+import io.pixelsdb.pixels.common.turbo.FunctionService;
+import io.pixelsdb.pixels.common.turbo.Invoker;
+import io.pixelsdb.pixels.common.turbo.InvokerProvider;
+import io.pixelsdb.pixels.common.turbo.WorkerType;
+import io.pixelsdb.pixels.common.utils.ConfigFactory;
+
+public class ScanStreamInvokerProvider implements InvokerProvider
```
[the remark this answers is not in the document]
Comment (bianhq) [reply pixels-turbo/pixels-invoker-vhive/src/main/java/io/pixelsdb/pixels/invoker/vhive/ScanStreamInvokerProvider.java:32]: use `@author` and `@create`.

**Review by bianhq — CHANGES_REQUESTED** [review]

**Review by bianhq — APPROVED** [review]
LGTM.

**Pull request merged**